File: gallicagram/update2023.py
import requests
from bs4 import BeautifulSoup
import os
page = requests.get(f"https://www.nytimes.com/search?dropmab=false&types=article&query=have&endDate=20231231&startDate=20230101")
soup = BeautifulSoup(page.content)
a = int(str(soup.find_all("script")[8]).split('totalCount":')[-1].split(",")[0])
os.system("head -n -1 base_presse_annees_nyt.csv > base_presse_annees_nyt.csv") ##Delete last line
os.system(f'echo 173,{a},2023' >> base_presse_annees_nyt.csv)

File: gallicagram/update2023.R
library(rvest)
library(stringr)
setwd("~/docker_gallicagram/gallicagram")
a = read_html("https://www.nytimes.com/search?dropmab=false&query=have&endDate=20231231&startDate=20230101&types=article")
count = str_split(html_text(a),'totalCount":')[[1]][2]
count = str_split(count,",")[[1]][1]
system("head -n -1 base_presse_annees_nyt.csv > temp.csv") ##Delete last line
system(str_c("echo 173,",count, ",2023 >> temp.csv"))
system("mv temp.csv base_presse_annees_nyt.csv")
#system("bash ~/executor")
